node/core/pvf/bin/puppet_worker.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! A simple binary that allows mocking the behavior of the real worker.
+
+use std::time::Duration;
+
+fn main() {
```

Comment:
Could we put this in some kind of macro or whatever? :P 

Cumulus currently uses the in-process capability for its integration test and it would be nice to create such a worker relative easy :D